### PR TITLE
Backend refactor

### PR DESCRIPTION
**Pull request by brantje**
- [x] Make use of Entity's
- [x] Make use of Mappers
- [x] Make use of services
- [x] Make use of an for each loop instead of join / group
- [x] Dump file sync for now (will be added later in a PR)

### CODE REVIEW

File: appinfo/autoload.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * Nextcloud - passman
```
Comment (enoch85) [inline appinfo/autoload.php]: Passman?

**Comment by enoch85** [comment]
> Dump file sync for now (will be added later in a PR)

Yes please :+1:

**Comment by enoch85** [comment]
Just tested this branch and got this:

```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'SELECT n.uid, n.id, n.name, n.grouping, n.shared, n.mtime, n.deleted, p.pid, GROUP_CONCAT(p.note SEPARATOR '') as note FROM oc_ownnote n INNER JOIN oc_ownnote_parts p ON n.id = p.id WHERE `uid` = ? and n.deleted = 0 GROUP BY p.id' with params ["daniel"]: SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #8 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'owncloud_db.p.pid' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```

**MySQL 5.7**

FYI.

**Comment by brantje** [comment]
Yea, already thought that the joins would cause trouble on some databases.
Guess i'll replace them with a foreach loop and a select.

**Comment by brantje** [comment]
Whoop scrutinizer is happy!

**Comment by enoch85** [comment]
@brantje Will this work or is it still deactivated?

**Pull request merged**